### DISCUSSION TITLE
Incremental Flow registration

### DISCUSSION THREAD

**Discussion by gatesn** [PrefectHQ/prefect#3357] [Q&A] [answered]
Suppose a Python repository that defines many flows, what's the recommended way to incrementally register them only when they've changed? 

In other words, how should I setup an idempotent CI/CD job to register all these flows without creating a new version on every CI run?

**Comment by cicdw** [comment, marked as answer]
Hi @gatesn - this is a great question; fundamentally, you only need to re-register whenever `flow.serialize()` changes, so if you had a way of tracking that value you could run a manual check.  We might consider exposing a built-in mechanism for this, but I'd want to think through the details to make sure it's used correctly.

One thought that comes to mind is to expose an `idempotency_key` in the GraphQL API for register Flows - `flow.register` could then use some deterministic hash of `flow.serialize()` as an idempotency key under-the-hood, and then you could safely call `flow.register` as many times as you want and Prefect would make sure new versions only get registered when they're necessary.

I'm open to other ideas as well!
